### ISSUE TITLE
ViewSet filter date picker missing margin

### ISSUE THREAD

**Issue by smark-1** [wagtail/wagtail#12299]
### Issue Summary
filtering a snippet by date shows the date field missing margins on one side.
![image](https://github.com/user-attachments/assets/89338c52-3e20-4b98-9ac0-2bc9c0008d2e)

### Steps to Reproduce
I am using the model BreadPage from the bakery demo as a ModelViewSet to demo this issue since the other viewsets do not make use of any date fields.

1. In bakery demo bakerydemo > breads > wagtail_hooks.py 

Add the following to the bottom
```python
from bakerydemo.breads.models import BreadPage
class ExampleModelViewSet(ModelViewSet):
    model = BreadPage
    add_to_admin_menu = True

    panels = [
        FieldPanel("title"),
    ]
    list_filter = ["last_published_at"]

register_snippet(ExampleModelViewSet)
```
2. in the admin sidebar click Bread Pages
3. Click filter options > Last published at. It will show below
![image](https://github.com/user-attachments/assets/01f4e069-8df0-4f4b-8715-528c5d1c2ad1)

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Wagtail version: 6.3 (dev)
### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by laymonage** [comment]
Thanks! I can confirm this issue. Seems to be more prominent in Firefox.

## Chrome

<img width="351" alt="image" src="https://github.com/user-attachments/assets/f4cccf94-ec68-48c5-bb74-0c1a55c2e11b">

## Firefox

<img width="385" alt="image" src="https://github.com/user-attachments/assets/a3207886-3312-4da9-8294-047f8599b759">

## Fix

The `<input>` element seems to overflow its parent wrapper. I think this can be fixed by adding `max-width: 100%` here:

https://github.com/wagtail/wagtail/blob/de2be5270e572e3a7be15c97083a76fee7bdf3a6/client/scss/components/forms/_input-text.scss#L42-L49

<img width="369" alt="image" src="https://github.com/user-attachments/assets/00534c5b-f88f-487e-a726-241fcdcfb7bb">

<img width="369" alt="image" src="https://github.com/user-attachments/assets/b3151030-6a7c-492c-8910-38edd23c63b2">

Marking this as a good first issue.

**Comment by Aditya8840** [comment]
Hi @laymonage, I would love to work this issue !!

**Comment by laymonage** [comment]
@Aditya8840 go right ahead :)

**Comment by Khizarshah01** [comment]
Hey @laymonage , can I work on this issue?

**Comment by zerolab** [comment]
@aditya-raj2230 there is already a pull request for it - https://github.com/wagtail/wagtail/pull/12336 - you could test it locally and help the quality assurance process

**Comment by TejasSaraf** [comment]
I want to work on this issue please assign me this issue.

**Comment by arsalancodegit** [comment]
i want to solve this issue kindly assign me this

**Comment by zerolab** [comment]
Fixed in #12347
